### PR TITLE
Layer Loading Queries

### PR DESCRIPTION
**Pull request by vaidotasp**
- Reworked queries for active features on user click.  'feature', 'csv', 'geojson', 'scene' layers are queried with `layer.queryFeatures` method. ALL OTHER layers are queried with regular `QueryTask`
- due to the fact that we can have layer or sublayer being active at any given time, implemented logic accounting for it in terms of `LayerSelector` and overall `DataPanel`.
- documented perf and usability issues in separate ticket.

### CODE REVIEW

File: src/js/components/leftPanel/dataPanel/DataTabView.tsx [hunk 1]
```diff
@@ -25,13 +26,18 @@ const DataTabView = (props: DataTabProps): JSX.Element => {
...
+    const activeLayerInfo = activeFeatures[activeFeatureIndex[0]];
 
+    //If layer has sublayers, we are using sublayerID to compare, otherwise it is layerID
+    function findLayer(f: LayerFeatureResult): boolean {
+      if (f.sublayerID) {
+        return String(f.sublayerID) === String(activeLayerInfo.sublayerID);
+      } else {
+        return String(f.layerID) === String(activeLayerInfo.layerID);
+      }
+    }
```
Comment (eyosef) [inline src/js/components/leftPanel/dataPanel/DataTabView.tsx]: I was thinking we could use a ternary operator?

```
const specificLayerID = f.sublayerID ? f.sublayerID : f.layerID

return String(f.specificLayerID) === String(activeLayerInfo.layerID);
```

File: src/js/components/leftPanel/dataPanel/DataTabView.tsx [hunk 2]
```diff
@@ -95,11 +101,16 @@ const DataTabView = (props: DataTabProps): JSX.Element => {
...
-          f => f.layerID === layerID
-        );
+      function handleLayerSwitch(id: string): void {
+        //If layer has sublayers, we are using sublayerID to compare, otherwise it is layerID
+        function findLayer(f: LayerFeatureResult): boolean {
+          if (f.sublayerID) {
+            return String(f.sublayerID) === id;
+          } else {
+            return String(f.layerID) === id;
+          }
```
Comment (eyosef) [inline src/js/components/leftPanel/dataPanel/DataTabView.tsx]: I was thinking we could use a ternary operator?

```
const specificLayerID = f.sublayerID ? f.sublayerID : f.layerID

return String(specificLayerID) === id
```

File: src/js/helpers/DataPanel.ts [hunk 3]
```diff
@@ -44,152 +17,150 @@ function esriQuery(url: string, queryParams: any): Promise<__esri.FeatureSet> {
...
+      if (
+        layer.type === 'feature' ||
+        layer.type === 'csv' ||
+        layer.type === 'geojson' ||
+        layer.type === 'scene'
+      ) {
+        try {
+          const featureResults = await layer.queryFeatures(queryParams);
+          //Ignore empty results
+          if (featureResults.features.length !== 0) {
```
Comment (eyosef) [inline src/js/helpers/DataPanel.ts]: affirmative check? Could help us ID edge-cases later on (if `features` is undefined, or is an unexpected result, for example)
`if (featureResults.features.length > 0) {`

**Comment by eyosef** [comment]
@vaidotasp what are your thoughts on using affirmative checks instead of `!==`? Just flagging as a general comment, feel free to let me know what you think!

**Comment by vaidotasp** [comment]
> @vaidotasp what are your thoughts on using affirmative checks instead of `!==`? Just flagging as a general comment, feel free to let me know what you think!

I have no real preference, so we can do positive checks for various "let's see if we have any features or data" functions. But sometimes we need negative one `if layer.type !== 'graphics'` stuff like that.

**Review by eyosef — COMMENTED** [review]
LGTM!

**Pull request merged**